### PR TITLE
Feature/have collection count type

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4382/

**Comment by jsteemann** [comment]
Sorry, but isn't RegisterCount just the same as RegisterId?

File: arangod/Aql/ConstrainedSortExecutor.h [hunk 1]
```diff
@@ -93,7 +93,7 @@ class ConstrainedSortExecutor {
   Fetcher& _fetcher;
   ExecutionState _state;
   size_t _returnNext;
-  std::vector<uint32_t> _rows;
+  std::vector<size_t> _rows;
```
Comment (jsteemann) [inline arangod/Aql/ConstrainedSortExecutor.h]: How many entries do we expect to have inside `_rows`?
Effectively doubling the size of each entry will obviously increase memory usage and will reduce cache karma.
If we are talking about few elements only, fine. But I am not sure. @danielhlarkin ?

Comment (danielhlarkin) [reply arangod/Aql/ConstrainedSortExecutor.h]: So this limits our queries to only having 4B rows at the intermediate step prior to limit. It doesn't seem wise to me. The standard sort executor uses `size_t` for similar purposes as well.

**Comment by dothebart** [comment]
yes, I've added `RegisterCount` so it can fit in semantically better in the places where I've added it.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4429/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4436/

**Comment by dothebart** [comment]
#4436 green.

**Pull request merged**